compiler/rustc_error_codes/src/error_codes/E0282.md

```diff
@@ -3,7 +3,7 @@ The compiler could not infer a type and asked for a type annotation.
 Erroneous code example:
 
 ```compile_fail,E0282
-let x = "hello".chars().rev().collect();
+let x;
```

Comment:
Can you find a better representative for `E0282`? This is trivially ambiguous, but I think this error should have a more descriptive example.

Also, the description below doesn't match this example.